src/ParallelAlgorithms/Events/ObserveAtExtremum.hpp

```diff
@@ -0,0 +1,412 @@
+    }
+    return (first[0] > second[0]) ? first : second;
+  }
+};
+
+}  // namespace detail
+
+namespace Events {
+/// @{
+/*!
+ * \brief Find the extremum of a Scalar<DataVector> over
+ * all elements, as well as the value of other functions
+ * at the location of that extremum (the reduction returns
+ * the minimum multiplied by (-1) for minima)
+ *
+ * \par Array sections
+ * This event supports sections (see `Parallel::Section`). Set the
+ * `ArraySectionIdTag` template parameter to split up observations into subsets
+ * of elements. The `observers::Tags::ObservationKey<ArraySectionIdTag>` must be
+ * available in the DataBox. It identifies the section and is used as a suffix
+ * for the path in the output file.
+ */
+template <typename ObservationValueTag, typename ObservableTensorTagsList,
+          typename NonTensorComputeTagsList = tmpl::list<>,
+          typename ArraySectionIdTag = void>
+class ObserveAtExtremum;
+
+template <typename ObservationValueTag, typename... ObservableTensorTags,
+          typename... NonTensorComputeTags, typename ArraySectionIdTag>
+class ObserveAtExtremum<ObservationValueTag,
+                        tmpl::list<ObservableTensorTags...>,
+                        tmpl::list<NonTensorComputeTags...>, ArraySectionIdTag>
+    : public Event {
+ private:
+  /// Reduction data will contain the time, and either the maximum
+  /// or the minimum of a function
+  using ReductionData = Parallel::ReductionData<
+      // Observation value
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      // Maximum of first component of a vector
+      Parallel::ReductionDatum<std::vector<double>,
+                               detail::MaxOfFirstElement>>;
+
+  /// Information about the scalar whose extremum we search for,
+  /// the type of extremum, and the other tensors to observer at
+  /// that extremum
+  struct ObserveTensors {
+    static constexpr Options::String help = {
+        "The tensor to extremize, and other tensors to observe."};
+
+    struct Name {
+      using type = std::string;
+      static constexpr Options::String help = {
+          "The name of the scalar to extremize."};
+    };
+
+    struct ExtremumType {
+      using type = std::string;
+      static constexpr Options::String help = {
+          "The type of extremum -- either Min or Max."};
+    };
+
+    struct AdditionalData {
+      using type = std::vector<std::string>;
+      static constexpr Options::String help = {
+          "List of other tensors to observe at the extremum"};
+    };
+
+    using options = tmpl::list<Name, ExtremumType, AdditionalData>;
+
+    ObserveTensors() = default;
+
+    ObserveTensors(const std::string in_scalar,
+                   const std::string in_extremum_type,
+                   const std::vector<std::string>& in_additional_data,
+                   const Options::Context& context = {});
+
+    std::string scalar_name, extremum_type;
```

Comment:
I don't know that we have an actual rule for this, but we generally put one member variable per line.